src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/OracleEngine.java

```diff
@@ -954,31 +944,21 @@ protected boolean checkConnection(final Connection conn) {
         }
     }
 
-    /**
-     * Gets the schema. This implementation uses the username when the schema is null.
-     *
-     * @return The schema or the username when the first is null.
-     */
-    @Override
-    protected String getSchema() {
-        return Optional.ofNullable(super.getSchema()).orElse(properties.getUsername());
```

Comment:
What happens now if we call getSchema and it is null?

Reply:
With these changes, we have 3 sources for schema information:
* _properties.getSchema()_ from the PDB properties, which can be null (meaning "use the default schema")
* _AbstractDatabaseEngine.getSchema()_ which returns the current schema from the Connection -> the current schema name or null if there is none (there should always be a schema, which is why the _AbstractDatabaseEngine.connect()_ method will throw an exception if the connection schema is null)
* it is assumed that once set right after connecting to the DB, the current schema won't change; the _AbstractDatabaseEngine.currentSchema_ will contain the name of that schema, so that it can be used when needed without having to query the DB to get it